modules/caddyhttp/celmatcher.go

```diff
@@ -150,7 +150,11 @@ func (m MatchExpression) Match(r *http.Request) bool {
 // UnmarshalCaddyfile implements caddyfile.Unmarshaler.
 func (m *MatchExpression) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 	for d.Next() {
-		m.Expr = strings.Join(d.RemainingArgs(), " ")
+		if d.CountRemainingArgs() > 1 {
+			m.Expr = strings.Join(d.RawRemainingArgs(), " ")
```

Comment:
Just curious why we need this condition as opposed to always joining the remaining args? If there's only one, shouldn't it just return it anyway?

Reply:
Because if you do Raw when it was quoted as a single token then you end up with the value being quoted wrapped which isn't valid CEL (try that change and run the tests to see if you don't follow what I mean)